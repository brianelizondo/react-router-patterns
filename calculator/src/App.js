import {React, useState } from 'react';
import { Routes, Route } from "react-router-dom";
import './App.css';

import Add from "./Add";
import Subtract from "./Subtract";
import Multiply from "./Multiply";
import Divide from "./Divide";

import Calculator from './Calculator';

function App() {
    return (
        // REACT ROUTER CALCULATOR WITH DIFFERENTS COMPONENTS
        // <div className="App">
        //     <Routes>
        //         <Route path="/add/:a/:b" element={ <Add /> } />
        //         <Route path="/subtract/:a/:b" element={ <Subtract /> } />
        //         <Route path="/multiply/:a/:b" element={ <Multiply /> } />
        //         <Route path="/divide/:a/:b" element={ <Divide /> } />
        //     </Routes>
        // </div>

        <div className="App">
            <Routes>
                <Route path="/:operation/:a/:b" element={ <Calculator /> } />
            </Routes>
        </div>
    );
}

export default App;
