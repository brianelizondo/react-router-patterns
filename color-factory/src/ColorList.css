.ColorList{
    text-align: center;
}
.ColorList-title{
    background-color: #333;
    color: white;
    font-size: 16px;
    font-weight: bold;
    padding: 40px 0px;
}
.ColorList-title a{
    color: white;
    font-size: 26px;
    text-decoration: underline;
}
.ColorList-title a:hover{
    color: aqua;
    text-decoration: none;
}

.ColorList-colors a{
    color: black;
    text-decoration: underline;
}
.ColorList-colors a:hover{
    color: aqua;
    text-decoration: none;
}