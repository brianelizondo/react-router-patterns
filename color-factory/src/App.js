import {React, useState } from 'react';
import { Routes, Route, Navigate } from "react-router-dom";
import './App.css';

import ColorList from "./ColorList";
import ColorDetails from "./ColorDetails";
import ColorNew from "./ColorNew";

function App() {
    const INITIAL_STATE = [];
    const [colors, setColors] = useState(INITIAL_STATE);

    const addColor = newColor => {
        setColors(colors => [newColor, ...colors]);
    };

    return (
        <div className="App">
            <Routes>
                <Route path="/colors/new" element={ <ColorNew addColor={addColor} /> } />
                <Route path="/colors/:color" element={ <ColorDetails colors={colors} /> } />
                <Route path="/colors" element={ <ColorList colors={colors} /> } />
            </Routes>
        </div>
    );
}

export default App;
