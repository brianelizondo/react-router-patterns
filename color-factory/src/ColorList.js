import React from "react";
import { Link } from "react-router-dom";
import "./ColorList.css"

function ColorList({ colors }){
    return (
        <div className="ColorList">
            <div className="ColorList-title">
                <div>Welcome to Color Factory</div>
                <div><Link to="/colors/new">Add a color</Link></div>
            </div>
            <div className="ColorList-colors">
                <div><h4>Please select a color:</h4></div>
                { colors.map(color => (<div key={ color.name.toLowerCase() }><Link to={ `/colors/${ color.name.toLowerCase() }` }>{ color.name }</Link></div>)) }
            </div>
        </div>
    );
}

export default ColorList;