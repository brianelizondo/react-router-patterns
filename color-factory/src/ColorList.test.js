import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import App from './App';
import ColorList from './ColorList';
const colorTest = [{
    name: "black",
    code: "#000000"
}];

// Smoke Tests
it("'ColorList' component renders without crashing", () => {
    render(<MemoryRouter><ColorList colors={colorTest} /></MemoryRouter>);
});

// Snapshot Tests
it("'ColorList' matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><ColorList colors={colorTest} /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// Others test
it("'ColorList' works listed the colors", () => {
    const { queryByText } = render(<MemoryRouter><ColorList colors={colorTest} /></MemoryRouter>);
  
    // expect list the colors
    expect(queryByText("Welcome to Color Factory")).toBeInTheDocument();
    expect(queryByText("black")).toBeInTheDocument();
});
