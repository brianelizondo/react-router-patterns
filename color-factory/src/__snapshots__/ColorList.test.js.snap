// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'ColorList' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorList"
  >
    <div
      class="ColorList-title"
    >
      <div>
        Welcome to Color Factory
      </div>
      <div>
        <a
          href="/colors/new"
        >
          Add a color
        </a>
      </div>
    </div>
    <div
      class="ColorList-colors"
    >
      <div>
        <h4>
          Please select a color:
        </h4>
      </div>
      <div>
        <a
          href="/colors/black"
        >
          black
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
